pkg/cmd/upgrade/upgrade_boot.go

```diff
@@ -0,0 +1,442 @@
+package upgrade
+
+import (
+	"fmt"
+	"github.com/jenkins-x/jx/pkg/auth"
+	"github.com/jenkins-x/jx/pkg/boot"
+	"github.com/jenkins-x/jx/pkg/cmd/helper"
+	"github.com/jenkins-x/jx/pkg/cmd/opts"
+	"github.com/jenkins-x/jx/pkg/cmd/templates"
+	"github.com/jenkins-x/jx/pkg/config"
+	"github.com/jenkins-x/jx/pkg/gits"
+	"github.com/jenkins-x/jx/pkg/kube"
+	"github.com/jenkins-x/jx/pkg/log"
+	"github.com/jenkins-x/jx/pkg/util"
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	"github.com/spf13/cobra"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+// UpgradeBootOptions options for the command
+type UpgradeBootOptions struct {
+	*opts.CommonOptions
+	Dir string
+}
+
+var (
+	upgradeBootLong = templates.LongDesc(`
+		This command creates a pr for upgrading a jx boot gitOps cluster, incorporating changes to the boot
+        config and version stream ref
+`)
+
+	upgradeBootExample = templates.Examples(`
+		# create pr for upgrading a jx boot gitOps cluster
+		jx upgrade boot
+`)
+)
+
+// NewCmdUpgradeBoot creates the command
+func NewCmdUpgradeBoot(commonOpts *opts.CommonOptions) *cobra.Command {
+	options := &UpgradeBootOptions{
+		CommonOptions: commonOpts,
+	}
+	cmd := &cobra.Command{
+		Use:     "boot",
+		Short:   "Upgrades jx boot config",
+		Long:    upgradeBootLong,
+		Example: upgradeBootExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			options.Cmd = cmd
+			options.Args = args
+			err := options.Run()
+			helper.CheckErr(err)
+		},
+	}
+	cmd.Flags().StringVarP(&options.Dir, "dir", "d", "", "the directory to look for the Jenkins X Pipeline and requirements")
+	return cmd
+}
+
+// Run runs this command
+func (o *UpgradeBootOptions) Run() error {
+	err := o.setupGitConfig(o.Dir)
+	if err != nil {
+		return errors.Wrap(err, "failed to setup git config")
+	}
+
+	if o.Dir == "" {
+		err := o.cloneDevEnv()
+		if err != nil {
+			return errors.Wrap(err, "failed to clone dev environment repo")
+		}
+	}
+
+	reqsVersionStream, err := o.requirementsVersionStream()
+	if err != nil {
+		return errors.Wrap(err, "failed to get requirements version stream")
+	}
+
+	upgradeVersionSha, err := o.upgradeAvailable(reqsVersionStream.URL, reqsVersionStream.Ref, "master")
```

Comment:
I am wondering if we should make `master` configurable. e.g. it can try to look up the latest `tag version` on versions repository.